src/cascadia/TerminalSettingsModel/Profile.cpp

```diff
@@ -247,10 +308,10 @@ void Profile::LayerJson(const Json::Value& json)
     JsonUtils::GetValueForKey(json, HiddenKey, _Hidden);
 
     // Core Settings
-    JsonUtils::GetValueForKey(json, ForegroundKey, _Foreground);
-    JsonUtils::GetValueForKey(json, BackgroundKey, _Background);
-    JsonUtils::GetValueForKey(json, SelectionBackgroundKey, _SelectionBackground);
-    JsonUtils::GetValueForKey(json, CursorColorKey, _CursorColor);
+    _Foreground.set = JsonUtils::GetValueForKey(json, ForegroundKey, _Foreground.setting);

```

Comment:
Do we want to file a follow-up for enlightening `JsonUtils` to be able to handle this case just as gracefully as the others?

Reply:
CC @DHowett 
I'm not sure what to do here. I've hit a few times where JsonUtils needed a little tweak. At the same time, they've been pretty small things so idk if complicating JsonUtils more would be the right approach. But at the same time, eh, it couldn't hurt?

Reply:
That was this discussion (transcribed from Teams markup...)

1. remove optional deduction from `Get/SetValue/ForKey`'s `Converter` detector
2. implement a partially specialized `ConversionTrait` over `std::optional<T>`
   1. `template<typename T> struct ConversionTrait<std::optional<T>>` may not even be legal code
   2. implement `FromJson` as return `nullopt`; or whatever if it's `null`
3. move the null detection logic from `GetValue` into `ConversionTrait<std::optional<T>>`
4. delete all of the `DeduceOptional` code
5. implement a `ConversionTrait<IReference<T>>` that is identical to the `std::optional<T>` version
6. mentally accept the fact that putting `null` in a non-nullable field will now cause a deserialization type error ("expected string, got null")
   1. mentally accept the fact that `std::optional<int>` will say, like, "expected null or number" which is cool
7. ------ NOW YOU CAN START TO FIX YOUR ORIGINAL PROBLEM -------
8. introduce `Settings::ExplicitlyNullable<T>` as `struct { bool _set; std::optional<T> _actualValue; }`
9. Implement `ConversionTrait<ExplicitlyNullable<T>>`
10. in `FromJson`, set `_set` even if it's null
11. then you can just check `_set`